packtools/sps/validation/article_toc_sections.py

```diff
@@ -2,58 +2,306 @@
+        }
+    }
+
+
+    Description automatically generated by ChatGPT.
+    """
+    article = _get_article_sections(sections)
+    sub_article = _get_sub_article_sections(sections)
+    article.update(sub_article)
+    return article
+
+
 class ArticleTocSectionsValidation:
     def __init__(self, xmltree):
         self.xmltree = xmltree
         self.article_toc_sections = ArticleTocSections(xmltree)
         self.article_titles = ArticleTitles(xmltree)
 
     def validate_article_toc_sections(self, expected_toc_sections):
-        resp = []
-        for lang, text in self.article_toc_sections.all_section_dict.items():
-            if text == self.article_toc_sections.article_section_dict.get(lang):
-                obj = 'article section title'
-            else:
-                obj = 'sub-article section title'
-            if text in (expected_toc_sections.get(lang) or []):
-                message = "OK, section titles match the document"
-                result = True
+        """
+        Check whether the TOC sections matches the options provided in a standard list.
+
+        XML input
+        ---------
+        <article xml:lang="es">
+        <front>
+            <article-meta>
+                <title-group>
+                    <article-title>Título do artigo em espanhol</article-title>
+                </title-group>
+                <article-categories>
+                    <subj-group subj-group-type="heading">
+                        <subject>Nome da seção do artigo em espanhol</subject>
+                    </subj-group>
+                </article-categories>
+            </article-meta>
+        </front>
+        <sub-article article-type="translation" xml:lang="en">
+            <front-stub>
+                <article-categories>
+                    <subj-group subj-group-type="heading">
+                        <subject>Nome da seção do sub-artigo em inglês</subject>
+                    </subj-group>
+                </article-categories>
+                <title-group>
+                    <article-title>HEDGING FUTURE CASH FLOWS WITH INTEREST-RATE FUTURES CONTRACTS: A DURATION AND
+                     CONVEXITY ANALYSIS UNDER THE NELSON SIEGEL MODEL</article-title>
+                </title-group>
+            </front-stub>
+        </sub-article>
+        </article>
+
+        Params
+        ------
+        expected_toc_sections : dict, such as:
+            {
+                "es": ["Nome da seção do artigo em espanhol"],
+                "en": ["Nome da seção do sub-artigo em inglês"]
+            }
+
+        Returns
+        -------
+        list of dict
+            A list of dictionaries, such as:
+            [
+                {
+                'title': 'Article section title validation',
+                'xpath': ".//article-meta//subj-group[@subj-group-type='heading']/subject",
+                'validation_type': 'match',
```

Comment:
@Rossi-Luciano o valor é 'value in list'

Reply:
A validação aqui deve ser dada uma lista de seções disponíveis nos idiomas correspondentes, verificar se as seções encontradas no XML estão presentes na lista no idioma correspondente.
Nota que apesar de falar em 'value in list', não necessariamente as seções passadas como parâmetro pode ser representadas com dicionário.

A idéia é que a função de validação receba como parâmetro TODAS as possíveis seções válidas com os idiomas correspondentes, por exemplo:
```python
sections = {"en": ["Articles", "Letters", "Editorial"], "es": ["Artículos", "Cartas", "Editorial"], "pt": ["Artigos", "Cartas", " "Editorial"]}
```

```python
sections = [{"language": "en", "titles": ["Articles", "Letters", "Editorial"]}, "es": {"language": "en", "titles": ["Articles", "Letters", "Editorial"]},]
```
ou outras representações

Reply:
OK, @robertatakenaka